File: fe/mobile/android/app/src/main/java/com/capstone/domain/repository/AuthRepository.java
package com.capstone.domain.repository;

import com.capstone.domain.model.UserProfile;

import androidx.annotation.Nullable;

public interface AuthRepository {
    UserProfile login(String accessToken) throws Exception;

    String sendOtp(String email) throws Exception;

    String verifyOtp(String email, String otp) throws Exception;

    String resetPassword(String email, String otp, String newPassword) throws Exception;

    String changePassword(String oldPass, String newPass, String confirmPass) throws Exception;

    UserProfile updateProfile(
            @Nullable String fullName,
            @Nullable String username,
            @Nullable String phoneNumber,
            @Nullable String birthdate,
            @Nullable String address,
            @Nullable Boolean gender
    ) throws Exception;

    UserProfile updateAvatar(byte[] imageBytes) throws Exception;

    UserProfile getMe() throws Exception;
}

File: fe/mobile/android/app/src/main/java/com/capstone/domain/repository/AuthRepository.kt
package com.capstone.domain.repository

import com.capstone.domain.model.UserProfile

interface AuthRepository {
    suspend fun login(accessToken: String): UserProfile
    suspend fun sendOtp(email: String): String
    suspend fun verifyOtp(email: String, otp: String): String
    suspend fun resetPassword(email: String, otp: String, newPassword: String): String
    suspend fun changePassword(oldPass: String, newPass: String, confirmPass: String): String
    suspend fun updateProfile(fullName: String?, username: String?, phoneNumber: String?, birthdate: String?, address: String?, gender: Boolean?): UserProfile
    suspend fun updateAvatar(imageBytes: ByteArray): UserProfile
    suspend fun getMe(): UserProfile
}